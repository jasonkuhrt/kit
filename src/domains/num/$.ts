/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
