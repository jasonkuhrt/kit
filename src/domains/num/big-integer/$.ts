export * as BigInteger from './$$.ts'
