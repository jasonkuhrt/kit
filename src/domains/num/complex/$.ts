export * as Complex from './$$.js'
