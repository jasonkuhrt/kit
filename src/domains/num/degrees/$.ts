// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
