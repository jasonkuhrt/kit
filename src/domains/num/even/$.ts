// Export the namespace containing all Even operations
export * as Even from './$$.ts'
