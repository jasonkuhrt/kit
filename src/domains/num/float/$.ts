// Export the namespace containing all Float operations
export * as Float from './$$.js'
