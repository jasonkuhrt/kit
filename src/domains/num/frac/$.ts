// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
