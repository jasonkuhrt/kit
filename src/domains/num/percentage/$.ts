// Export the namespace containing all Percentage operations
export * as Percentage from './$$.ts'
