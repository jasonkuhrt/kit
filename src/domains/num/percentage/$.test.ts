import { Test } from '#test'
import { Percentage } from './$.js'

Test.describe('is')
  .on(Percentage.is)
  .cases(
    // Valid cases - values between 0 and 1 inclusive
    [0],
    [0.5],
    [1],
    [0.25],
    [0.75],
    [0.001],
    [0.999],
    // Invalid cases - values outside 0-1 range
    [-0.1],
    [1.1],
    [2],
    [-1],
    [100],
    [Infinity],
    [NaN],
    ['0.5'],
    [null],
  )
  .test()

Test.describe('from')
  .on(Percentage.from)
  .cases(
    // Valid cases
    [0],
    [0.5],
    [1],
    [0.25],
    [0.75],
    [0.001],
    [0.999],
    // Invalid cases - should throw
    [-0.1],
    [1.1],
    [2],
    [-1],
    [100],
  )
  .test()

Test.describe('tryFrom')
  .on(Percentage.tryFrom)
  .cases(
    // Valid cases
    [0],
    [0.5],
    [1],
    [0.25],
    [0.75],
    [0.001],
    [0.999],
    // Invalid cases - should return null
    [-0.1],
    [1.1],
    [2],
    [-1],
    [100],
  )
  .test()
