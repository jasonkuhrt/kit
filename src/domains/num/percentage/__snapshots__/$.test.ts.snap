// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be between 0 and 1, got: -0.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be between 0 and 1, got: -1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.001) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.001
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.001
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.5
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.25) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.25
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.25
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.75) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.75
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.75
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0.999) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.999
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.999
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.1
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be between 0 and 1, got: 1.1
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
2
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be between 0 and 1, got: 2
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be between 0 and 1, got: 100
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is("0.5") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.001) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.001
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.25) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.25
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.75) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.75
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0.999) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.999
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.1
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
2
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-0.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-0.1
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-1
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.001) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.001
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.001
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.5) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.5
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.5
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.25) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.25
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.25
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.75) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.75
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.75
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0.999) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0.999
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0.999
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.1) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.1
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(2) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
2
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(100) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
100
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;
